xapian-core/cluster/cluster.cc

```diff
@@ -604,3 +612,42 @@ Cluster::Internal::recalculate()
     }
     centroid.divide(size());
 }
+
+StemStopper::StemStopper(const Stem &stemmer_, stem_strategy strategy)
+    : stem_action(strategy), stemmer(stemmer_)
+{
+    LOGCALL_CTOR(API, "StemStopper", stemmer_ | strategy);
+}
+
+string
+StemStopper::get_description() const
+{
+    string desc("Xapian::StemStopper(");
+    unordered_set<string>::const_iterator i;
+    for (i = stop_words.begin(); i != stop_words.end(); ++i) {
+	if (i != stop_words.begin()) desc += ' ';
+	desc += *i;
+    }
+    desc += ')';
+    return desc;
+}
+
+void
+StemStopper::add(const string &term)
+{
+    LOGCALL_VOID(API, "StemStopper::add", term);
+    switch (stem_action) {
+	case STEM_NONE:
+	    stop_words.insert(term);
+	    break;
+	case STEM_ALL_Z:
+	    stop_words.insert('Z' + stemmer(term));
```

Comment:
Yes I had the same tests for STEM_ALL and STEM_ALL_Z. I'll add those changes in. I hadn't run a make check before pushing.